src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -115,19 +117,28 @@ final case class ThermalGrid(
   def updateState(
       tick: Long,
       state: ThermalGridState,
+      lastAmbientTemperature: Temperature,
       ambientTemperature: Temperature,
       qDot: Power,
   ): (ThermalGridState, Option[ThermalThreshold]) = if (qDot > zeroKW)
-    handleInfeed(tick, ambientTemperature, state, qDot)
+    handleInfeed(tick, lastAmbientTemperature, ambientTemperature, state, qDot)
   else
-    handleConsumption(tick, ambientTemperature, state, qDot)
+    handleConsumption(
+      tick,
+      lastAmbientTemperature,
+      ambientTemperature,
+      state,
+      qDot,
+    )
 
   /** Handles the case, when a grid has infeed. First, heat up all the houses to
     * their maximum temperature, then fill up the storages
     * @param tick
     *   Current tick
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
```

Comment:
Maybe to add some specificity:

```suggestion
    *   Ambient temperature valid up until (not including) the current tick
```